inc/Plugin.php

```diff
@@ -97,21 +115,45 @@ function ( $project_id, $originals_added, $originals_existing, $originals_obsole
 			function( $translation_set_id ) {
 				/* @var GP_Translation_Set $translation_set */
 				$translation_set = GP::$translation_set->get( $translation_set_id );
-				$last_modified   = $translation_set->last_modified();
 
-				if ( $last_modified ) {
-					$last_modified = new DateTime( $last_modified, new DateTimeZone( 'UTC' ) );
-				} else {
-					$last_modified = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
+				$zip_provider = new ZipProvider( $translation_set );
+				$zip_provider->generate_zip_file();
+			}
+		);
+
+		add_filter(
+			'gp_update_meta',
+			function( $meta_tuple ) {
+				$project = ( new ProjectLocator( $meta_tuple['object_id'] ) )->get_project();
```

Comment:
The check for the `$allowed_keys` should come before checking for the project as it seems to be less expensive.